### PR TITLE
[IMP] mass_mailing: Migrate mailing contact list/kanban to owl

### PR DESCRIPTION
**Pull request by tivisse**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mass_mailing/static/src/views/mass_mailing_views.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates id="mailing_contact_views" xml:space="preserve">
+    <t t-name="MailingContactListView.buttons" t-inherit="web.ListView.Buttons" t-inherit-mode="primary" owl="1">
+        <xpath expr="//t[@t-if='props.showButtons']">
```
Comment (awa-odoo) [inline addons/mass_mailing/static/src/views/mass_mailing_views.xml:4]: No "position" here?

Button seems there anyway, I tested, but not sure how it works, maybe there is a default.

File: addons/mass_mailing/static/src/views/mailing_contact_view_list.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        this.actionService = useService("action");
+    }
+
+    onImport() {
+        const context = this.props.context;
+        const actionParams = { context: context };
+        if (!context.default_mailing_list_ids && context.active_model === 'mailing.list' && context.active_ids) {
+            actionParams.context.default_mailing_list_ids = context.active_ids;
+        }
+        this.actionService.doAction('mass_mailing.mailing_contact_import_action', actionParams);
```
Comment (awa-odoo) [inline addons/mass_mailing/static/src/views/mailing_contact_view_list.js:26]: I'm not sure this part still works properly.

To test:
- Email Marketing app
- Mailing Lists -> Mailing Lists
- Click on a record (e.g: "Interested in Tree Promotions")
- Click on the "Recipients" stat button
- Click on import

Master:
image

Your branch:
image

(Did not test on kanban but I'm guessing it also has the issue).

File: addons/mass_mailing/static/src/views/mailing_contact_view_kanban.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+/** @odoo-module **/
+
+import { KanbanController } from "@web/views/kanban/kanban_controller";
+import { kanbanView } from '@web/views/kanban/kanban_view';
+import { registry } from '@web/core/registry';
+import { useService } from "@web/core/utils/hooks";
+
+export class MailingContactController extends KanbanController {
```
Comment (awa-odoo) [inline addons/mass_mailing/static/src/views/mailing_contact_view_kanban.js:8]: I've noticed than now the handlers like "onImport" seem to be placed on the Renderer in owl.

But it seems to work fine on the Controller too, not sure if it matters.

**Review by awa-odoo — COMMENTED** [review]
Hello @tivisse 

Code is looking good, thanks!
Added some minor comments.

However, it seems like the "default mailing list" feature is broken.
Maybe I did not test correctly, can you take a look?

Cheers 🏩

File: addons/mass_mailing/static/src/views/mass_mailing_views.xml (hunk 1 above)
Comment (tivisse) [reply addons/mass_mailing/static/src/views/mass_mailing_views.xml:4]: Indeed there's a default value, but I'll provide it explicitly :)

**Review by awa-odoo — APPROVED** [review]
Thanks @tivisse !

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**